setup.py

```diff
@@ -21,13 +21,13 @@
             'lexibank=pylexibank.commands',
         ],
         'cldfbench.scaffold': [
-            'lexibank=pylexibank.scaffold:LexibankTemplate'
+            'lexibank_simple=pylexibank.scaffold:LexibankTemplate'
         ],
     },
     platforms='any',
     python_requires='>=3.5',
     install_requires=[
-        'cldfbench[excel]>=0.4',
+        'cldfbench[excel]>=0.5',
```

Comment:
@tresoldi you mean pinning minor versions, a la `~=0.5`?

Reply:
We discussed pinning elsewhere, here's the outcome (AFAIR):

- Pinning means datasets will cease to "work" when a newer version of the pinned library is installed - even if this wouldn't break anything.
- Not pinning means breaking changes could go undetected for some time: Touch a dataset after a long time - and travis may complain, cause it may install a newer version of some req.

So to get the some of the best of both worlds, we decided
- no pinning
- record the last working configuration in `cldf/requirements.txt`

This means in the scenario above, where a dataset "breaks" on travis without anything in the dataset having changed, you'd start investigating by comparing `cldf/requirements.txt` with what travis got.

It's still a trade-off, but from past experience should be the least-effort solution.

Reply:
More than pinning (which I understand as a particular version, but correct me if you mean something different), I meant specifying the range of versions, either with a wildcard (i.e., `==0.5.*`) or with a range (i.e., `>=0.5,<0.6`).

My idea is that, if a new version of one the libraries breaks any of the released datasets, it should get a higher revision number. The idea would be to guarantee that a virtual environment with lexibank with always compile all the datasets in there, while still allowing to fix bugs and to introduce new features that are not backwards incompatible. We had people installing a new, entire lexibank ecosystem for each dataset, manually pinning the versions and making debugging hard...

But I see the point of your trade-off, after all a new version (say, a backwards incompatible `cldfbench` 1.0) would likely be part of a release party where all packaged depending on it would be updated as well. We should at least instruct people to update everything when it is needed. Or maybe just adding an ugly

```
import pkg_resources
from subprocess import call

packages = [dist.project_name for dist in pkg_resources.working_set]
call("pip install --upgrade " + ' '.join(packages), shell=True)
call("lexibank makecldf all")
```
auxiliary script?

Reply:
You can also pin "compatible versions", see https://www.python.org/dev/peps/pep-0440/#compatible-release
But this still means that `clldutils` 3.x broke many datasets - although none of the backwards incompatible changes in it affected the code in the datasets.

Reply:
It seems the best practice regarding backwards incompatible changes nowadays is a good deprecation policy - and downstream users taking deprecation warnings seriously.